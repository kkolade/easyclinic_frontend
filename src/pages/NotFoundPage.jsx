import {
  Center, Container, Flex, Text, Title, rem,
} from '@mantine/core';
import { IconChevronLeft } from '@tabler/icons-react';

import RouterButton from 'components/RouterButton';

const NotFoundPage = () => (
  <Container fluid>
    <Center h="100vh">
      <Flex direction="column" justify="center" align="center" rowGap="sm">
        <Title order={2} align="center">
          Not Found (404)
        </Title>
        <Text c="dimmed" fz={rem(17)} align="center">
          Sorry, the page you are looking for does not exist.
        </Text>
        <RouterButton to="/" sx={{ marginTop: rem(16) }}>
          <IconChevronLeft size={rem(20)} />
          <Text fz={rem(15)} sx={{ marginLeft: rem(8) }}>
            Back to home
          </Text>
        </RouterButton>
      </Flex>
    </Center>
  </Container>
);

export default NotFoundPage;
