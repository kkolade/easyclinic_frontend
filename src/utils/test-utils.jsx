/* eslint-disable import/no-extraneous-dependencies */

import { MantineProvider } from '@mantine/core';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { MemoryRouter as Router } from 'react-router-dom';
import store from '../redux/store';
import theme from '../theme';

const renderWithProviders = ({ children }) => (
  <Provider store={store}>
    <MantineProvider withNormalizeCSS withGlobalStyles theme={theme}>
      <Router>{children}</Router>
    </MantineProvider>
  </Provider>
);

const customRender = (ui, options) => render(ui, { wrapper: renderWithProviders, ...options });

// re-export everything
export * from '@testing-library/react';

// override render method
export { customRender as render };
