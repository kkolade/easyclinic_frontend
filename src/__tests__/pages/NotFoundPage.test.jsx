import NotFoundPage from 'pages/NotFoundPage';
import { render } from 'test-utils';

describe('NotFoundPage', () => {
  it('should render correctly', () => {
    const { asFragment } = render(<NotFoundPage />);
    expect(asFragment()).toMatchSnapshot();
  });
});
