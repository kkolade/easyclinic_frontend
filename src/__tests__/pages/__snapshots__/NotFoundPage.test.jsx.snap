// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFoundPage should render correctly 1`] = `
<DocumentFragment>
  <div
    class="mantine-Container-root mantine-6jkqqi"
  >
    <div
      class="mantine-Center-root mantine-1x99tkp"
    >
      <div
        class="mantine-qtk8ny"
      >
        <h2
          class="mantine-Text-root mantine-Title-root mantine-ci3f5c"
        >
          Not Found (404)
        </h2>
        <div
          class="mantine-Text-root mantine-xheqkd"
        >
          Sorry, the page you are looking for does not exist.
        </div>
        <a
          class="mantine-UnstyledButton-root mantine-Button-root mantine-1cflbei"
          data-button="true"
          href="/"
          type="button"
        >
          <div
            class="mantine-1wpc1xj mantine-Button-inner"
          >
            <span
              class="mantine-1ryt1ht mantine-Button-label"
            >
              <svg
                class="tabler-icon tabler-icon-chevron-left"
                fill="none"
                height="1.25rem"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="1.25rem"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M15 6l-6 6l6 6"
                />
              </svg>
              <div
                class="mantine-Text-root mantine-3y9j5z"
              >
                Back to home
              </div>
            </span>
          </div>
        </a>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
